libraries/AP_HAL_Linux/Scheduler.cpp

```diff
@@ -66,6 +66,29 @@ extern const AP_HAL::HAL& hal;
 Scheduler::Scheduler()
 { }
 
+
+void Scheduler::init_realtime()
+{
+#if APM_BUILD_TYPE(APM_BUILD_Replay)
+    // we don't run Replay in real-time...
+    return;
+#elif APM_BUILD_TYPE(APM_BUILD_UNKNOWN)
```

Comment:
this will break examples that actually want to have realtime priorities.  What's the problem with running as root?  We can fix it to actually check for CAP_SYS_NICE.  Or try and provide a fallback.